### PR TITLE
Create elections list and view from election dates

### PR DESCRIPTION
**Pull request by LindsayYoung**
Bases the election year off of dates.
We were missing special elections before.
We can't use the candidate history directly because people file for the wrong election.
The date data starts in 2003.

### CODE REVIEW

**Comment by LindsayYoung** [comment]
closes #2476

**Comment by noahmanger** [comment]
Will we need to make any front-end changes for this?

**Comment by LindsayYoung** [comment]
The only change should be limiting the year select, I will test it with the webapp tomorrow

**Comment by LindsayYoung** [comment]
I am making better tables, there is still a disconnect between that and the results, that I am working on

**Comment by LindsayYoung** [comment]
We don't have Senate class data in the elections data so that makes it so that we can't  figure out the senate incumbent for an upcoming special election with this data

**Comment by LindsayYoung** [comment]
Got it working better for Senate but need to circle back on presidential-  seeing duplicates

File: data/sql_updates/election_outcome.sql [hunk 1]
```diff
@@ -1,45 +1,160 @@
...
+    where office_sought != 'P'
+        group by
+        election_state,
+        office_sought,
+        election_district,
+        election_yr
+),
+election_dates_president as(
+    select distinct on (election_yr)
+        'US'::char(2) as election_state,
```
Comment (ccostino) [inline data/sql_updates/election_outcome.sql]: I'm sure this is okay, but just wanted to double check - substituting in `US` here as a state abbreviation won't have an adverse effect on the data returned or any processing that takes place leveraging this view as the source, right?

File: data/sql_updates/election_outcome.sql [hunk 2]
```diff
@@ -1,45 +1,160 @@
...
-union all
-(
-    select distinct on (cand_office_st, cand_office_district, cand_election_yr)
-        cand_election_yr - 2 as election_yr, *
-    from disclosure.cand_valid_fec_yr
+),
+-- This creates election records and the incumbent info if applicable.
+-- We need to do this because a null in the elections date district data
+-- means all the districts in the state but the district is listed,
+-- it only applies to one district.
```
Comment (ccostino) [inline data/sql_updates/election_outcome.sql]: Thank you for all of these comments describing what each subset of data represents, it's a huge help! :-)

File: data/sql_updates/election_outcome.sql [hunk 3]
```diff
@@ -1,45 +1,160 @@
...
+        ed.election_district as cand_office_district,
+        fec.cand_pty_affiliation,
+        fec.cand_name,
+        ed.election_yr::numeric as election_yr
+    from election_dates_house_senate ed
+    left join
+        incumbent_info fec on
+        ed.election_state = fec.cand_office_st and
+        ed.office_sought = fec.cand_office and
+        -- in the date table the district is 3 characters
```
Comment (ccostino) [inline data/sql_updates/election_outcome.sql]: And thank you for noting data oddities/inconsistencies as well!

File: data/sql_updates/election_outcome.sql (hunk 1 above)
Comment (ccostino) [inline data/sql_updates/election_outcome.sql]: We just caught up with an in-person chat and this mirrors the output from before, so not only is it okay it's expected and other code is set to support it already.  We're good!

**Pull request merged**